src/quotes_handler.rs

```diff
@@ -237,18 +248,32 @@ pub async fn integrity(
                 .await;
         };
 
+        let ima_ml_entry = req.uri().query().unwrap();
+        let nth_entry = match ima_ml_entry.find("ima_ml_entry=") {
+            None => 0,
+            Some(idx) => {
+                let number = &ima_ml_entry[idx + 13..];
+                let last: usize =
+                    number.find(|c: char| !c.is_numeric()).unwrap_or(0);
```

Comment:
Shouldn't this be `.unwrap_or(number.len())`, if there are only numeric characters?

Reply:
I tried this now but I end up with this error message (patch pushed): 
```
$ cargo build
   Compiling keylime_agent v0.1.0 (/home/stefanb/dev/rust-keylime)
error[E0596]: cannot borrow data in an `Arc` as mutable
   --> src/quotes_handler.rs:271:63
    |
271 |             read_measurement_list(&ima_ml_pathbuf, nth_entry, &mut data.ima_measurement_list)?;
    |                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
    |
    = help: trait `DerefMut` is required to modify through a dereference, but it is not implemented for `Arc<QuoteData>`

For more information about this error, try `rustc --explain E0596`.
error: could not compile `keylime_agent` due to previous error
```

I am not sure how do resolve this...